src/coreclr/jit/instrsxarch.h

```diff
@@ -697,22 +697,22 @@ INST1(fld,              "fld",              IUM_WR, 0x0000D9,
-INST1(setge,            "setge",            IUM_WR, 0x0F009D,                                                            INS_FLAGS_ReadsFlags)
-INST1(setle,            "setle",            IUM_WR, 0x0F009E,                                                            INS_FLAGS_ReadsFlags)
-INST1(setg,             "setg",             IUM_WR, 0x0F009F,                                                            INS_FLAGS_ReadsFlags)
+INST1(seto,             "seto",             IUM_WR, 0x0F0090,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setno,            "setno",            IUM_WR, 0x0F0091,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setb,             "setb",             IUM_WR, 0x0F0092,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setae,            "setae",            IUM_WR, 0x0F0093,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(sete,             "sete",             IUM_WR, 0x0F0094,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setne,            "setne",            IUM_WR, 0x0F0095,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setbe,            "setbe",            IUM_WR, 0x0F0096,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(seta,             "seta",             IUM_WR, 0x0F0097,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(sets,             "sets",             IUM_WR, 0x0F0098,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setns,            "setns",            IUM_WR, 0x0F0099,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setp,             "setp",             IUM_WR, 0x0F009A,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setnp,            "setnp",            IUM_WR, 0x0F009B,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setl,             "setl",             IUM_WR, 0x0F009C,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setge,            "setge",            IUM_WR, 0x0F009D,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setle,            "setle",            IUM_WR, 0x0F009E,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
+INST1(setg,             "setg",             IUM_WR, 0x0F009F,                                                            INS_FLAGS_ReadsAllFlagsExceptAF)
```

Comment:
This in particular (along with the `jcc` and `cmovcc` groups) is what I called out here: https://github.com/dotnet/runtime/issues/53053#issuecomment-846134204

That is, while `setcc`, `jcc`, and `cmovcc` as a "whole" can read any flag; the individual variants only read a subset and I could have set these more explicitly.

This should be:
* `seto` and `setno` are only `INS_FLAGS_ReadsOF`
* `setb` and `setae` are only `INS_FLAGS_ReadsCF`
  * `setc`/`setnae` and `setnc`/`setnb`  are the respective aliases
* `sete` and `setne` are only `INS_FLAGS_ReadsZF`
  * `setz` and `setnz`  are the respective aliases
* `setbe` and `seta` are both `INS_FLAGS_ReadsCF` and `INS_FLAGS_ReadsZF`
  * `setna` and `setnbe` are the respective aliases
* `sets` and `setns` are only `INS_FLAGS_ReadsSF`
* `setp` and `setnp` are only `INS_FLAGS_ReadsPF`
  * `setpe` and `setpo` are the respective aliases
* `setl` and `setge` are both `INS_FLAGS_ReadsSF` and `INS_FLAGS_ReadsOF`
  * `setnge` and `setnl` are the respective aliases
* `setle` and `setg` are `INS_FLAGS_ReadsZF`, `INS_FLAGS_ReadsSF` and `INS_FLAGS_ReadsOF`
  * `setng` and `setnle` are the respective aliases

Reply:
Yes, I need to add that information as well.

Reply:
I added this information for `jcc`, `cmovcc` and `setcc`. Also included minimal flags to indicate that the instruction resets one or more flags.